src/Lecture1.hs

```diff
@@ -123,4 +140,12 @@ and lower than 6 elements (4, 5, 6, 7, 8 and 9).
 
 🕯 HINT: Use recursion to implement this function.
 -}
-lowerAndGreater n list = error "TODO"
+lowerAndGreater:: Int -> [Int] -> String
+lowerAndGreater n list = show n ++ " is greater than " ++ show ( go [0,0] list !! 1) ++ " elements" ++ " and lower than " ++ show (head (go [0,0] list)) ++ " elements"
```

Comment:
That's correct! 💯 

However, with this solution, you will walk through the list twice.
You can write it more performant with only one traversal of the list 🙂
It's desirable to write functions that traverse lists only a single time and consume elements only once. In that case, the composition of such functions will have nice streaming and lazy properties.

In this case, it's enough to save the result of `go` in a local variable

Reply:
Thank you for this amazing course, I did done a star on main repo a long ago, but got time to start course only now. Oh, and thanks for review, I did changed it now but I don't quite sure how it works, is it okay to submit new iterations in same PR?

Reply:
Sure! Feel free to update the PR here after the review :)

The PR looks good now, feel free to merge it 🚢